packages/sku/src/services/vite/plugins/skuViteMiddlewarePlugin.ts

```diff
@@ -0,0 +1,43 @@
+import type { SkuContext } from '@/context/createSkuContext.js';
+import type { Plugin } from 'vite';
+import { createRequire } from 'node:module';
+
+const require = createRequire(import.meta.url);
+
+export const skuViteMiddlewarePlugin = (skuContext: SkuContext): Plugin => ({
+  name: 'vite-plugin-sku-server-middleware',
+  configureServer(server) {
+    return () => {
+      server.middlewares.use(async (req, res, next) => {
+        const host = req.headers.host;
+        const hostname = host?.split(':')[0];
+        const site =
+          skuContext.sites.find((skuSite) => skuSite.host === hostname) || '';
+
+        const isHtml = req.url === '/index.html';
+        if (isHtml) {
+          const renderEntry = require.resolve('../entries/vite-render.jsx');
+          const clientEntry = require.resolve('../entries/vite-client.jsx');
+
+          const { viteRender } = await server.ssrLoadModule(renderEntry);
+
+          const renderedHtml = await viteRender({
```

Comment:
`viteRender` has type `any` here. I think this is a spot where a typecast would prevent headaches. Having a centralized `Render` type would help keep these in sync.